Terragrunt website jekyll gh

tomaszantas opened the pull request:
The Terragrunt website built with Jekyll.

The website is compatible with Github Pages and can be served from the **/docs** folder on **master** branch as it is. No more actions are required. Github will automatically rebuild the website after every commit.

brikis98 commented on CONTRIBUTING.md:
```diff
@@ -1,8 +1,8 @@
 # Contribution Guidelines
 
-Contributions to this repo are very welcome! We follow a fairly standard [pull request 
+Contributions to this repo are very welcome! We follow a fairly standard [pull request
```
NIT: were there any changes to this file or just whitespace? If the latter, could you revert, as this PR is already big enough!

tomaszantas replied:
A new one section was added, but the rest are whitespaces. I cleaned commits, and ignore those whitespaces in new commits.

brikis98 commented on config/config_helpers.go:
```diff
@@ -39,7 +39,6 @@ var TERRAFORM_COMMANDS_NEED_VARS = []string{
 	"plan",
 	"push",
 	"refresh",
-	"validate",
```
Unintentional change?

tomaszantas replied:
That's right. I synced original repo to my fork and it had a commit: 
https://github.com/gruntwork-io/terragrunt/commit/4f7fcb6d1ca0914589e0145e7d53f0721ac1ff5e

I rebase my branch and I ignored this file.

tomaszantas commented:
I rebased branch and split changes into 2 commits:

1. https://github.com/gruntwork-io/terragrunt/pull/976/commits/31c0995dcafcf2431142ddb4f20bced4283a4751 - commits only `/docs` folder containing Terragrunt website. It changes nothing in existing files.

2. https://github.com/gruntwork-io/terragrunt/pull/976/commits/ab15c598cc1dedae57bb5d7e04b793abe1816639 - contains only change in `CONTRIBUTING.md`. It's a link to the website documentation.

brikis98 commented on CONTRIBUTING.md:
```diff
@@ -36,6 +36,9 @@ Development](http://tom.preston-werner.com/2010/08/23/readme-driven-development.
 stays up to date and allows you to think through the problem at a high level before you get lost in the weeds of 
 coding.
 
+The documentation is built with Jekyll and hosted on the Github Pages from `docs` folder on `master` branch. Check out
```
Thanks for updating the docs!

brikis98 commented on docs/README.md:
```diff
@@ -0,0 +1,421 @@
+
+## Run
+
+1. Go to the `docs` folder:
```
NIT: First step should be to install Ruby / Bundler on your OS.

brikis98 commented on docs/README.md:
````diff
@@ -0,0 +1,421 @@
+```
+$ git add .
+$ git commit -m "New version"
+$ git push origin <branch>
````
Are you suggesting that after a build, you have to commit the generated code? That shouldn't be necessary with GitHub Pages, as it'll run the Jekyll build for you.

tomaszantas replied:
I forgot to update this section with the last commit. Now it is:

```
# Deployment

GitHub Pages automatically rebuilds the website from the `/docs` folder on the `master` branch.
```

brikis98 commented on docs/README.md:
```diff
@@ -0,0 +1,421 @@
+2. Edit content.
+3. If you changed, added or removed the file title or any heading, [update navigation](#updating-navigation).
+
+## B) Add a new page
```
Thank you for the detailed docs 👍

brikis98 commented on docs/_data/navigation-use-cases-no-children.yml:
```diff
@@ -0,0 +1,17 @@
+name: use cases na children
```
na children?

brikis98 replied:
Oh, was this supposed to be "no children"?

tomaszantas replied:
You are right. It was a typo.

brikis98 commented on docs/_documentation/02_features/built-in-functions.md:
```diff
@@ -0,0 +1,437 @@
+
+  - [get\_aws\_caller\_identity\_arn()](#get_aws_caller_identity_arn)
+
+  - [get\_aws\_caller\_identity\_user\_id()](#get_aws_caller_identity_user_id)
```
Sanity check: are the `\` there to escape the `_`, so Markdown doesn't interpret that as italics?

tomaszantas replied:
Correct, `_` makes it italics.

brikis98 commented on docs/_documentation/02_features/built-in-functions.md:
```diff
@@ -0,0 +1,437 @@
+      |     └── terragrunt.hcl
+      └── terragrunt.hcl
+
+Imagine `terragrunt/mysql/terragrunt.hcl` and `terragrunt/secrets/mysql/terragrunt.hcl` include all settings from the root `terragrunt.hcl` file:
```
Sanity check: GitHub shows this all in Italics. Bug or rendering issue?

tomaszantas replied:
Github makes it italics because of `*` (star) in: `└── *.tf`. I added an escape character `\` so it should fix the problem.

brikis98 commented on docs/_includes/styles.html:
```diff
@@ -0,0 +1,3 @@
+<!-- <link rel="stylesheet" href="{{site.baseurl}}{{ site.assets_base_url }}css/app.css" /> -->
```
Sanity check: should this be here, commented out?

tomaszantas replied:
I removed Less from the project with a new commit, so it's no longer relevant.

brikis98 commented on docs/assets/css/less/bootstrap/dist/js/npm.js:
```diff
@@ -0,0 +1,13 @@
+require('../../js/popover.js')
+require('../../js/scrollspy.js')
+require('../../js/tab.js')
+require('../../js/affix.js')
```
Are we expected to run some manual step to concat the JS code? Or are all these part of bootstrap and we're using the already-concatenated version?

tomaszantas replied:
No js script from bootstrap was used, so I deleted everything. Nothing requires additional building.

tomaszantas commented on docs/assets/css/app.less:
```diff
@@ -0,0 +1,38 @@
+# 2. Make changes in "assets/css/less/*" files
+# 3. Build project with "JEKYLL_ENV=production bundle exec jekyll build"
+# 4. Copy "_site/assets/css/app.css" to the "assets/css/app_less.css"
+# 5. Switch back from "jekyll" to "github-pages" in Gemfile
```
I converted Less to Scss through CSS. The direct conversion brought me problems with overriding variables. But the final result is that a developer doesn't have to take extra steps to update styles and can use Scss. So the above comment in the `app.less` is no longer relevant.

brikis98 replied:
Wonderful, thank you!
